ext/opcache/ZendAccelerator.c

```diff
@@ -118,6 +118,29 @@ zend_bool file_cache_only = 0;  /* process uses file cache only */
 zend_bool fallback_process = 0; /* process uses file cache fallback */
 #endif
 
+#ifdef HAVE_SIGPROCMASK
+static sigset_t mask_all_signals;
+# define BLOCK_ALL_SIGNALS() \
+	sigset_t oldmask; \
+	MASK_ALL_SIGNALS()
+
+# ifdef ZTS
+#  define MASK_ALL_SIGNALS() \
+	tsrm_sigmask(SIG_BLOCK, &mask_all_signals, &oldmask)
+#  define UNBLOCK_ALL_SIGNALS() \
+	tsrm_sigmask(SIG_SETMASK, &oldmask, NULL)
+# else
+#  define MASK_ALL_SIGNALS() \
+	sigprocmask(SIG_BLOCK, &mask_all_signals, &oldmask)
+#  define UNBLOCK_ALL_SIGNALS() \
+	sigprocmask(SIG_SETMASK, &oldmask, NULL)
+# endif
+
+#else
+# define BLOCK_ALL_SIGNALS()   do {} while(0)
+# define UNBLOCK_ALL_SIGNALS() do {} while(0)
+#endif
```

Comment:
I'd suggest to keep this functionality in Zend and under the previous macro names, so it remains usable in 3rd party extensions. apcu at least uses them.

Reply:
It's a good suggestion.

Reply:
Please see the updated PR now.